### DISCUSSION TITLE
Access Attachments

### DISCUSSION THREAD

**Discussion by NatroNx** [RogerSelwyn/O365-HomeAssistant#258] [Q&A] [answered]
Hello,

I'm trying do download attachments to a specific folder to let paperless consume it. 

How to I access the actual attachments of a mail? I see the sensor I've created but no Idea how to get the actual attachment

ty

**Comment by RogerSelwyn** [comment]
To be honest I've never really looked at what download attachments does in any detail, I've passed across the code of course since I took the integration, but not focused on it. All it does is extract the file name from the attachment and shows it on the entity, it does not make the attachment available.

**Reply by NatroNx** [reply]
Oh, I see. I got confused by the "download_attachments: False"  Option.

Ok I'll search for another solution. Thanks for your response

**Reply by RogerSelwyn** [reply]
I think it would be possible to add the storage of attachments, I'd put them in o365_storage folder, by date and time somehow. 

I'd have to change the download_attachments default to False, because it would mean people tidying up the attachment downloaded or it would fill their storage up.

By the way, I'd only add this to the MS365_Mail integration which is taking over (along with the related integrations) from O365 as the go forward integration set.

**Comment by RogerSelwyn** [comment, marked as answer]
I’ve added the functionality to this release of MS365-Mail - https://github.com/RogerSelwyn/MS365-Mail/releases/tag/v0.2.0

**Reply by NatroNx** [reply]
this is great and works fine for me.

So this is now my workaround for Paperless NGX as well.  Paperless still can't use a modern authentication and actually I don't want to have multiple services / addons to fiddle around in my mailbox :) 

So for all desperate paperless users out there - use this addon and get your attachments in a defined folder. Then copy / move this file into he paperless consume folder and you're done.

I did it with a simple node red copy job. Can probably done within HomeAssistant as well but as I use nodeRed to do other file related things it was a good fit to do it there.
